### PR TITLE
(#381) Update queryStringToFormObject utility function

### LINKED ISSUE #381: Update queryStringToFormObject utility function
Utility function `queryStringToFormObject` is currently implemented in file `utilities/queryStringToFormObject.js`

Update this utility function:

**Parent Issue:** #362

> **ESTIMATE** 5

### Sub-Tasks
- [x]  Rename to `queryStringToSearchCriteria` (this includes 12 test files)
- [ ] Refactor useAppInitializer to move queryStringToSearchCriteria logic into trial description page and trial results page (useEffect and/or useCallback)
- [ ]  Pull the methods out of redux store
- [ ]  Make everything redux-independent
- [ ]  Dictate the shape of the generated form object
- [ ] Update unit tests for utility function
- [ ] Update the following integration test (/AdvancedSearchPage/LocationSection/LocationByZipcode.feature)
```gherkin
Scenario: Negative: User is not able to search for an incorrect zip code but is able to search for empty zip

Scenario: Negative: User is not able to search for an incorrect zip code AND is NOT able to search for empty zip
        Given the user navigates to "/about-cancer/treatment/clinical-trials/search/advanced"
        Then the page title is "Find NCI-Supported Clinical Trials"
        When user selects "ZIP Code" radio button
        When user types "999g9" in "U.S. ZIP Code" field
        Then alert "Please enter a valid 5 digit U.S. zip code" is displayed
        And user clicks on "Find Trials" button
        Then the search is not executed and path is "/about-cancer/treatment/clinical-trials/search/advanced"
        When user clears "U.S. ZIP Code" input field
        And user clicks on "Find Trials" button
        Then text "Sorry, you seem to have entered invalid criteria. Please check the following, and try your search again:" is displayed
        And the invalid criteria table displays the following
            | Criteria |
            | Zip      |
        And text "For assistance, please contact the Cancer Information Service. You can chat online or call 1-800-4-CANCER (1-800-422-6237)." is displayed
        And "chat online" has a link "https://livehelp.cancer.gov/app/chat/chat_landing"
        And "Try a new search" link has a href "/about-cancer/treatment/clinical-trials/search/advanced"
```
* this will act as a contract with the react-form search page object, they should match up
* Once this function is updated, we need to update everywhere that is calling this function, in order to make sure the entire app doesn't break

## NOTES:
This task should be updating https://github.com/NCIOCPL/clinical-trials-search-app/blob/develop/src/utilities/queryStringToFormObject.js. It looks like there need to be a few things taken into consideration:
1. `import { defaultState } from '../store/reducers/form';` will have to change because the reducer is going away.
    * this should be a copy of https://github.com/NCIOCPL/clinical-trials-search-app/blob/develop/src/store/reducers/form.js 
    * Remove:
       ```
        isDirty: false, // only updated after submission of either form
        zipModified: false,
        hasInvalidZip: false, // zip does not return coodinates
        refineSearch: false, //is the form in refine search mode
        ageModified: false,
        cancerTypeModified: false,
        subtypeModified: false,
        stagesModified: false,
        keywordPhrasesModified: false,
       ```       
2. The shape of the Object returned may not be the cleanest?
     * The unit tests are setup for that shape, so major changes might not be worth it.
3. The Label fetchers and Zip Code lat/long fetchers probably are ok, they are passed into the function anyway, and this should stay this way. (results and description page will need to pass in the fetchers and where they get them from is TBD and not part of this.)

### PR DESCRIPTION
**Pull request by seyilonge-nci**
Closes #381

* Rename to queryStringToSearchCriteria
* Updated `queryStringToSearchCriteria`
* Moved `queryStringToSearchCriteria` call from `useAppInitializer` to `ResultsPage`
* Moved `runQueryFetchers` into `queryStringToSearchCriteria`
* Updated `TrialDescriptionPage` to use `queryStringToSearchCriteria`object in place of form object in redux store
* Updated unit test for `queryStringToSearchCriteria` (WIP...)
* Restored deleted unit test files
* Further unit tests for `queryStringToSearchCriteria` (WIP...)
* Reverted changes to `queryStringToSearchCriteria` and `useStoreToFindTrials`
* Added `runQueryFetchers` utility which is called from respective pages for `queryStringToSearchCriteria`
* Added store action to update store form with search criteria
* Updated unit tests
* Added mock data files for integration tests
* Removed `hasInvalidZip` from default state, commented out int-test block
* Updated `ResultsPage` to display error for invalid zip
* Updated `TrialDescriptionPage` to use form store values for "formType" and "isDirty"
* Updated integration test with refined acceptance criteria
* Removed `nihOnly` references as it's not handled in `queryStringToSearchCriteria` and it's conditional check in `SearchCriteriaTable.jsx` hinders the search criteria table from being displayed
* Removed `hasInvalidZip` from `defaultStateCopy` for unit tests
* Displaying invalid zip error message using `ErrorPage`, updated proptype for ErrorPage to resolve warning

### CODE REVIEW

File: src/views/TrialDescriptionPage/TrialDescriptionPage.jsx [hunk 1]
```diff
@@ -96,7 +115,7 @@ const TrialDescriptionPage = () => {
 		event: `ClinicalTrialsSearchApp:Other:ShareButton`,
 		analyticsName,
 		linkName: 'UnknownLinkName',
-		formType: formType,
+		formType: searchCriteriaObject?.formType,
```
Comment (adriancofie) [inline src/views/TrialDescriptionPage/TrialDescriptionPage.jsx:118]: 👀 Fancy  👍

**Comment by bryanpizzillo** [comment]
@seyilonge what happened to the tests that were deleted?

**Comment by seyilonge-nci** [comment]
> @seyilonge what happened to the tests that were deleted?

Shoot! File rename snafu. Tests will be back in next commit. Thanks!

**Comment by bryanpizzillo** [comment]
@seyilonge when you get to repushing this, can you fix the commit message to have the proper ticket number please?

**Comment by seyilonge-nci** [comment]
> @seyilonge when you get to repushing this, can you fix the commit message to have the proper ticket number please?

Sorry missed that. Fixed!

File: src/utilities/__tests__/queryStringToFormObject.adv.loc.negative.js [hunk 2]
```diff
@@ -1,139 +0,0 @@
-import { queryStringToFormObject } from '../queryStringToFormObject';
-
```
Comment (KateMashkinaNIH) [inline src/utilities/__tests__/queryStringToFormObject.adv.loc.negative.js:2]: should this be deleted? Or renamed?

**Review by KateMashkinaNIH — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Comment by bryanpizzillo** [comment]
@arcepaul to create a ticket for the at NIH fix
@kate-mashkina is going to create a ticket for the zipcode issue

Close both of those referering this ticket as the fix.

**Comment by sarinapadilla** [comment]
@bryanpizzillo ticket created for zipcode issue: https://github.com/NCIOCPL/clinical-trials-search-app/issues/428

It's been immediately closed out referencing this ticket as the fix!

**Pull request merged**